src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java

```diff
@@ -140,18 +142,22 @@ public void process(PackageNode packageNode) {
                     DataMapperStructureVisitor visitor = new DataMapperStructureVisitor(structureMap);
                     visitor.visit(recordTypeDef);
                     String typeName = null;
+                    String serialized = null;
                     if (recordTypeDef.symbol instanceof  BRecordTypeSymbol) {
                         typeName = ((BRecordTypeSymbol) recordTypeDef.symbol).toString();
+                        String encodedTypeName = typeName;
+                        if (!hasAcceptedCharacters(encodedTypeName)) {
```

Comment:
Do we need to encode this conditionally? IINM if the string doesn't contain any charactor that does not support in a URL then the original string will not change when we URLEncode it.

Reply:
I removed the conditional URL encoding logic and updated the code.